File: app/src/main/java/com/lucasurbas/counter/data/repository/CounterRepository.java
package com.lucasurbas.counter.data.repository;

import com.lucasurbas.counter.data.model.Counter;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Observable;
import io.reactivex.Single;

public interface CounterRepository {

    Single<Counter> getCounterWithId(int id);

    Observable<Counter> getCounterUpdatesWithId(int id);

    Single<List<Counter>> getCounters();

    Observable<List<Counter>> getCountersUpdates();

    Completable updateCounter(Counter counter);

}

File: app/src/main/java/com/lucasurbas/counter/data/repository/CounterRepository.kt
package com.lucasurbas.counter.data.repository

import com.lucasurbas.counter.data.model.Counter

import io.reactivex.Completable
import io.reactivex.Observable
import io.reactivex.Single

interface CounterRepository {

    fun getCounterWithId(id: Int): Single<Counter>

    fun getCounterUpdatesWithId(id: Int): Observable<Counter>

    fun getCounters(): Single<List<Counter>>

    fun getCountersUpdates(): Observable<List<Counter>>

    fun updateCounter(counter: Counter): Completable

}
